### PR TITLE
A number of autospec improvements

### PR DESCRIPTION
**Pull request by bryteise**
Taken from a patchset from @fenrus75

### CODE REVIEW

File: autospec/pkg_integrity.py [hunk 1]
```diff
@@ -223,19 +223,21 @@ def head_request(url):
...
 def get_signature_url(package_url):
-    if '://pypi.' in package_url[:13]:
+    if 'samba.org' in package_url:
+        return package_url + '.asc'
+    elif '://pypi.' in package_url[:13]:
         return package_url + '.asc'
     elif 'mirrors.kernel.org' in package_url:
         return package_url + '.sig'
     else:
         try:
```
Comment (busykai) [inline autospec/pkg_integrity.py]: This bothers me so much, maybe it's the appropriate moment to re-write the whole `try:` body as:
```python
try:
    iter = (package_url + "." + ext for ext in ("sig", "asc", "sign"))
        for sign_url in iter:
            if head_request(sign_url) in (200, 302):
                return sign_url
except:
    pass
```

**Comment by matthewrsj** [comment]
Commit 1: I'd rather see this consistently implemented throughout the code base before applying this. I don't think applying this selectively helps with readability.

**Comment by matthewrsj** [comment]
Commit 2: +1

**Comment by matthewrsj** [comment]
Commits 3-6: +1

**Comment by matthewrsj** [comment]
Commit 7: -1, look just a few lines lower. This is implemented already.

File: autospec/pkg_integrity.py (hunk 1 above)
Comment (matthewrsj) [reply autospec/pkg_integrity.py]: Agree that is much better, but maybe not as part of this patchset. Feel free to open a PR for that change though.

**Comment by matthewrsj** [comment]
Commits 8 and 11: +1

Commits 9 and 10 require further review.

File: autospec/pkg_integrity.py (hunk 1 above)
Comment (busykai) [reply autospec/pkg_integrity.py]: Opened #128. It can wait till this merges so @bryteise wouldn't have to resolve.

**Comment by bryteise** [comment]
@matthewrsj For commit 1, do you want the types literally everywhere or just for global values before accepting? I'm dropping the patch from this series though in the mean time.

**Comment by bryteise** [comment]
@matthewrsj Once 9 and 10 are reviewed I'll push and updated set with 1 and 7 removed.

**Comment by matthewrsj** [comment]
I guess just for all global values would be a good start, though I'm not entirely convinced these are necessary at all. From what I can tell it is just for readability and linting warnings, no? Anyways that can be in a separate PR to discuss there.

> we need to start the typing somewhere tbh

Ok, but let's at least cover all of *something* not just some of the globals. At least cover all the globals before merging this.

**Comment by bryteise** [comment]
@fenrus75 I do note that the makes autospec python 3.6+ which we've stepped back from before.

**Comment by matthewrsj** [comment]
> @fenrus75 I do note that the makes autospec python 3.6+ which we've stepped back from before.

Good point, we deprecated the new format strings (`f"..."`) to support at least 3.5

File: autospec/specfiles.py [hunk 2]
```diff
@@ -431,8 +436,11 @@ def write_variables(self):
                 flags.extend(["-Os", "-ffunction-sections", "-fdata-sections", "-fno-semantic-interposition"])
         if config.config_opts['security_sensitive']:
             flags.append("-fstack-protector-strong")
+            flags.append("-mzero-caller-saved-regs")
```
Comment (matthewrsj) [inline autospec/specfiles.py]: This change is not related to the AVX512 code this patch applies.

File: autospec/specfiles.py [hunk 3]
```diff
@@ -581,6 +594,14 @@ def write_cmake_install(self):
             self._write_strip("popd")
             self._write_strip("rm -f %{buildroot}/usr/bin/*")
 
+        if config.config_opts['use_avx512']:
+            self._write_strip("mkdir -p %{buildroot}/usr/lib64/haswell/avx512_1")
+            self._write_strip("pushd clr-build-avx512")
+            self._write_strip("%s %s || :\n" % (self.install_macro, self.extra_make_install))
+            self._write_strip("mv %{buildroot}/usr/lib64/*so* %{buildroot}/usr/lib64/haswell/avx512_1 || :")
+            self._write_strip("popd")
```
Comment (matthewrsj) [inline autospec/specfiles.py]: optional, can this be moved one line up?

File: autospec/specfiles.py [hunk 4]
```diff
@@ -838,6 +885,35 @@ def write_autogen_pattern(self):
                                       config.extra_configure32))
             self.write_make_line(True)
             self._write_strip("popd")
+
+        if config.config_opts['use_avx2']:
```
Comment (matthewrsj) [inline autospec/specfiles.py]: Not really part of AVX512, but fine because it is the same type of change.

File: autospec/specfiles.py [hunk 5]
```diff
@@ -1169,8 +1275,32 @@ def write_meson_pattern(self):
         self.write_variables()
         self._write_strip('CFLAGS="$CFLAGS" CXXFLAGS="$CXXFLAGS" LDFLAGS="$LDFLAGS" meson --prefix /usr --buildtype=plain {0} builddir'.format(config.extra_configure))
         self._write_strip("ninja -v -C builddir")
+
+        if config.config_opts['32bit']:
```
Comment (matthewrsj) [inline autospec/specfiles.py]: Here and below, different change from AVX512 changes.

File: autospec/specfiles.py (hunk 3 above)
Comment (bryteise) [reply autospec/specfiles.py]: You want the popd moved up one line? Do you want that moved up in the avx2 code above this as well?

**Comment by matthewrsj** [comment]
Commit 10: Eh, sure. I'm not sure I like the fallback being the homedir but I don't really have a better alternative, so +1 for that.

File: autospec/specfiles.py (hunk 3 above)
Comment (matthewrsj) [reply autospec/specfiles.py]: sure, optional like I said though. little deal.

Comment (bryteise) [reply autospec/specfiles.py]: Skipping for now.

**Comment by bryteise** [comment]
@matthewrsj Updated.
